pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReader.java

```diff
@@ -34,10 +34,16 @@
 public interface RecordReader extends Closeable {
 
   /**
-   * Initializes the record reader when needed
+   * initializing recordreader with inputpath, schema and recordreader config. <br/>
+   * The implementation can chose to ignore one or more of these parameters and handle null gracefully <br/>
+   *
+   * @param inputPath absolute path to the file/directory
+   * @param schema Pinot Schema associated with the table
+   * @param recordReaderConfig config for the reader specific to the format. e.g. delimiter for csv format etc
+   * @throws Exception if the arguments are invalid
    */
-  void init(SegmentGeneratorConfig segmentGeneratorConfig)
-      throws IOException;
+  void init(@Nullable String inputPath, @Nullable Schema schema, @Nullable RecordReaderConfig recordReaderConfig)
```

Comment:
Why not move input path and schema into RecordReaderConfig? That way, everything that the record reader needs is there in the config.

Reply:
This config is for record readers with extra settings (e.g. CSV delimiter), which can be null. Input path and schema are required for all record readers

Reply:
I thought about it. Not all RecordReader need that. Ideally, we want to pass a PinotFS interface wrapper but want to avoid making too many changes in this PR. My first goal is to decouple the modules without adding too much functionality/changing existing classes.  We will have a better idea when we move the interfaces to spi module.

Reply:
> This config is for record readers with extra settings (e.g. CSV delimiter), which can be null. Input path and schema are required for all record readers

Not really, some of them take GenericRows. There are too many implementations for recordReader: used in testing, segment conversion, Minion etc. Most of the readers get what they want in their constructors today.